framework/src/utils/RandomGenerator.C

```diff
@@ -0,0 +1,45 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "RandomGenerator.h"
+#include  <random>
+
+class MersenneTwister
+{
+public:
+  std::mt19937 _mt_eng;
+};
```

Comment:
So, for this PR I'd say use `MooseRandom`. Later we can switch it to C++11 if we wanted to. This way the RNG will be consistent. What do you think @permcody ?